app/controllers/contributors_controller.rb

```diff
@@ -0,0 +1,190 @@
+
+  # PUT /plans/:plan_id/contributors/:id
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  def update
+    authorize @plan
+    args = translate_roles(hash: contributor_params)
+    args = process_org(hash: args)
+    args = process_orcid_for_update(hash: args)
+
+    if @contributor.update(args)
+      redirect_to edit_plan_contributor_path(@plan, @contributor),
+                  notice: success_message(@contributor, _("saved"))
+    else
+      flash.now[:alert] = failure_message(@contributor, _("save"))
+      render :edit
+    end
+  end
+  # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
+
+  # DELETE /plans/:plan_id/contributors/:id
+  def destroy
+    authorize @plan
+    if @contributor.destroy
+      msg = success_message(@contributor, _("removed"))
+      redirect_to plan_contributors_path(@plan), notice: msg
+    else
+      flash.now[:alert] = failure_message(@contributor, _("remove"))
+      render :edit
+    end
+  end
+
+  private
+
+  def contributor_params
+    base_params = %i[name email phone org_id org_name org_crosswalk]
+    role_params = Contributor.new.all_roles
+
+    params.require(:contributor).permit(
+      base_params,
+      role_params,
+      identifiers_attributes: %i[id identifier_scheme_id value attrs]
+    )
+  end
+
+  # Translate the check boxes values of "1" and "0" to true/false
+  def translate_roles(hash:)
+    roles = Contributor.new.all_roles
+    roles.each { |role| hash[role.to_sym] = hash[role.to_sym] == "1" }
+    hash
+  end
+
+  # Convert the Org Hash into an Org object (creating it if necessary)
+  # and then remove all of the Org args
+  def process_org(hash:)
+    return hash unless hash.present? && hash[:org_id].present?
+
+    org = org_from_params(params_in: hash, allow_create: true)
+    hash = remove_org_selection_params(params_in: hash)
+    return hash unless org.present?
+
+    hash[:org_id] = org.id
+    hash
+  end
+
+  # When creating, just remove the ORCID if it was left blank
+  def process_orcid_for_create(hash:)
+    return hash unless hash[:identifiers_attributes].present?
+
+    id_hash = hash[:identifiers_attributes][:"0"]
+    return hash unless id_hash[:value].blank?
+
+    hash.delete(:identifiers_attributes)
+    hash
+  end
+
+  # When updating, destroy the ORCID if it was blanked out on form
+  def process_orcid_for_update(hash:)
+    return hash unless hash[:identifiers_attributes].present?
+
+    id_hash = hash[:identifiers_attributes][:"0"]
+    return hash unless id_hash[:value].blank?
+
+    existing = @contributor.identifier_for_scheme(scheme: "orcid")
+    existing.destroy if existing.present?
+    hash.delete(:identifiers_attributes)
+    hash
+  end
+
+  # =============
+  # = Callbacks =
+  # =============
+  def fetch_plan
+    @plan = Plan.includes(:contributors).find_by(id: params[:plan_id])
+    return true if @plan.present?
+
+    redirect_to root_path, alert: _("plan not found")
+  end
```

Comment:
I really like this style of fetching data.  It would be nice to convert some of the older controllers to conform to this style rather than fetching/validating the data per-action